### PR TITLE
Feature new sdk login usecase logout usecase

### PR DESCRIPTION
**Pull request by xurxodev**
create use cases according to Clean Architecture

### CODE REVIEW

File: app/src/main/java/org/eyeseetea/malariacare/LoginActivity.java [hunk 1]
```diff
@@ -69,6 +59,10 @@
     private AlertDialog alertDialog;
     private static final String TAG = "LoginActivity";
 
+    public IUserAccountRepository mUserAccountRepository = new UserAccountRepository(this);
+    public LoginUseCase mLoginUseCase = new LoginUseCase(mUserAccountRepository);
+    public LogoutUseCase mLogoutUseCase = new LogoutUseCase(mUserAccountRepository);
```
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/LoginActivity.java]: LogoutUseCase is never used, do you think it is necessary?

File: app/src/main/java/org/eyeseetea/malariacare/data/database/datasources/UserAccountLocalDataSource.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        User user = new User(credentials.getUsername(), credentials.getUsername());
+
+        Session.setUser(user);
+
+        saveCredentials(credentials);
+
+        callback.onSuccess(null);
+    }
+
+    private void saveCredentials(Credentials credentials) {
```
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/datasources/UserAccountLocalDataSource.java]: @xurxodev Do you think we should delete this [not used method] (https://github.com/EyeSeeTea/malariapp/blob/ab2bba337eeb22bb5d81337d16987d095f517bfa/app/src/main/java/org/eyeseetea/malariacare/LoginActivity.java#L212) in loginActivity ? I think saveUserDetails is no longer needed if we have saveCredentials

File: app/src/main/java/org/eyeseetea/malariacare/domain/exception/NetworkException.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package org.eyeseetea.malariacare.domain.exception;
+
+
+public class NetworkException extends Exception {
+    public NetworkException() {
+        super("Credentials not valid");
```
Comment (idelcano) [inline app/src/main/java/org/eyeseetea/malariacare/domain/exception/NetworkException.java]: @xurxodev can you revise the exception message? thanks!

**Comment by xurxodev** [comment]
@idelcano changes ready, sorry I not update commit message

**Pull request merged**